src/main/java/events/writer/PubSubEventWriter.java

```diff
@@ -0,0 +1,172 @@
+        try {
+            publisherBuilder = Publisher.newBuilder(topicName)
+                    .setCredentialsProvider(() -> GoogleCredentials.fromStream(getCredentials(serviceAccountPath)));
+            // Configure channel proxy is only for this to work in Vodafone VPC
+            configureChannelProxy(publisherBuilder, proxyHost, proxyPort);
+            this.publisher = publisherBuilder.build();
+            logger.info("Publisher created successfully");
+        } catch (IOException e) {
+            logger.error("Error creating pubsub events.publisher. Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     *
+     * @param projectId
+     * @param topicId
+     */
+    public PubSubEventWriter(String projectId,  String topicId) {
+        Publisher.Builder publisherBuilder = null;
+        this.publisher = null;
+
+        TopicName topicName = TopicName.of(projectId, topicId);
+        try {
+            publisherBuilder = Publisher.newBuilder(topicName);
+            this.publisher = publisherBuilder.build();
+            logger.info("Publisher created successfully");
+        } catch (IOException e) {
+            logger.error("Error creating pubsub events.publisher. Error: " + e.getMessage());
+        }
+    }
+
+    /**
+     *
+     * @param saPath
+     * @return
+     * @throws FileNotFoundException
+     */
+    private InputStream getCredentials(String saPath) throws FileNotFoundException {
+        File credentialsFile = new File(saPath);
+        return new FileInputStream(credentialsFile);
+    }
+
+    /**
+     *
+     * @param builder
+     * @param proxyHost
+     * @param proxyPort
+     * @throws NumberFormatException
+     */
+    private void configureChannelProxy(Publisher.Builder builder, String proxyHost, String proxyPort) throws NumberFormatException {
+        SocketAddress proxySocketAddress = new InetSocketAddress(proxyHost, Integer.parseInt(proxyPort));
+        builder.setChannelProvider(
+                InstantiatingGrpcChannelProvider.newBuilder()
+                        .setChannelConfigurator(managedChannelBuilder -> managedChannelBuilder.proxyDetector(new ProxyDetector() {
+                            @Nullable
+                            @Override
+                            public ProxiedSocketAddress proxyFor(SocketAddress socketAddress) {
+                                if (socketAddress == null) return null;
+
+                                return HttpConnectProxiedSocketAddress.newBuilder()
+                                        .setTargetAddress((InetSocketAddress) socketAddress)
+                                        .setProxyAddress(proxySocketAddress)
+                                        .build();
+                            }
+                        }))
+                        .build()
+        );
+    }
+
+    /**
+     *
+     * @param event
+     */
+    public void publishEvent(Event event) {
+        Gson gson = new Gson();
```

Comment:
Best practice for Gson is to create a static variable and re-use it in the methods. This will avoid multiple (potentially costly) instantiations.